packages/evm/src/fee/PercentageFee.ts

```diff
@@ -0,0 +1,59 @@
+import { FeeHandlerType } from '@buildwithsygma/core';
+import { PercentageERC20FeeHandlerEVM__factory } from '@buildwithsygma/sygma-contracts';
+import { utils } from 'ethers';
+
+import type { EvmFee } from '../types.js';
+
+import type { EvmFeeCalculationParams } from './types.js';
+import { BaseEvmTransferFeeCalculator } from './types.js';
+
+export class PercentageFeeCalculator extends BaseEvmTransferFeeCalculator {
+  constructor() {
+    super();
+  }
+
+  async calculateFee(params: EvmFeeCalculationParams): Promise<EvmFee> {
+    if (params.feeHandlerType === FeeHandlerType.PERCENTAGE) {
+      const percentageFeeHandlerContract = PercentageERC20FeeHandlerEVM__factory.connect(
+        params.feeHandlerAddress,
+        params.provider,
+      );
+
+      const calculatedFee = await percentageFeeHandlerContract.calculateFee(
+        params.sender,
+        params.sourceSygmaId,
+        params.destinationSygmaId,
+        params.resourceSygmaId,
+        params.depositData ?? utils.formatBytes32String(''),
+        utils.formatBytes32String(''),
+      );
+
+      const feeBounds = await percentageFeeHandlerContract._resourceIDToFeeBounds(
+        params.resourceSygmaId,
+      );
+
+      const feePercentage = (
+        await percentageFeeHandlerContract._domainResourceIDToFee(
+          params.destinationSygmaId,
+          params.resourceSygmaId,
+        )
+      ).toNumber();
+      const percentage = feePercentage / (await percentageFeeHandlerContract.HUNDRED_PERCENT());
```

Comment:
nit: add a line here for consistency

Reply:
also, nit: I would evaluate the async expression first and assigned to a var and then use it in the division